include/s3select_oper.h

```diff
@@ -243,17 +243,36 @@ struct binop_mult {
 struct binop_div {
     double operator()(double a,double b)
     {
-        return a / b;
+        if (b == 0)
+            {
+                base_s3select_exception("division by zero is not allowed");
+                return -1;
+            } else {
+                return a / b;
+            }
     }
 };
 
 struct binop_pow {
     double operator()(double a,double b)
     {
         return pow(a,b);
-    }
+    } 
 };
 
+struct binop_modulo {
+    int64_t operator()(int64_t a,int64_t b)
+    {
+        if (b == 0)
+            {
+                base_s3select_exception("Mod zero is not allowed");
+                return -1;
+            } else {
+                return (a % b + b) % b;
+            }
```

Comment:
Hi @galsalomon66 Should we take into account of negative integers?

Reply:
i got the point (its a good point), as for negative.
it increase number of instructions (comparing to a%b), 
our operator is aligned with C % operator, its better to let the user decide what to do with negative result.

Reply:
ok, I will reverse the change.

Reply:
Updated. Removed support to negative integers.